### PR TITLE
Refactor markdown math render, add dollor-backquote syntax support

### PR DESCRIPTION
**Pull request by wxiaoguang**
Add more tests

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: web_src/css/repo/home-file-list.css [hunk 1]
```diff
@@ -41,7 +41,7 @@
   display: flex;
   align-items: center;
   gap: 0.5em;
-  padding: 6px 10px;
+  padding: 8px 10px;
```
Comment (silverwind) [inline web_src/css/repo/home-file-list.css]: Same change is included in https://github.com/go-gitea/gitea/pull/32835 and cleaned up, you should remove it here.

Comment (wxiaoguang) [reply web_src/css/repo/home-file-list.css]: But that PR is still in progress and needs more time. https://github.com/go-gitea/gitea/pull/32835#issuecomment-2542817028

I hope I can have this change as soon as possible.

Comment (silverwind) [reply web_src/css/repo/home-file-list.css]: Then delete line 44, it's already covered by line 32 because the selector is the same.

Comment (wxiaoguang) [inline web_src/css/repo/home-file-list.css]:
```suggestion
```

Comment (wxiaoguang) [reply web_src/css/repo/home-file-list.css]: Done a8b53985d44e788aa6155e886fdd1c27b19b4e66

**Comment by silverwind** [comment]
BTW I would prefer to move the `display` class to a `data-display` attribute.

**Review by silverwind — APPROVED** [review]

**Comment by wxiaoguang** [comment]
> BTW I would prefer to move the `display` class to a `data-display` attribute but that would require changes to the markup sanitizer, so not so simple.

Hmm yes, I'd like to avoid touching the sanitizer too much. So maybe a 2-3 words name is better, easier to search globally.

**Pull request merged**